### PR TITLE
Write Metadata for Trained Models to the Metastore

### PR DESCRIPTION
**Pull request by waltaskew**

### CODE REVIEW

File: usl_models/usl_models/flood_ml/metastore.py [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+    Returns:
+      The document ID of the written model metadata document.
+    """
+    # Use the GCS location of the model as a unique value for the ID.
+    # URL-escape the ID, as characters such as / are not allowed in document IDs.
+    model_id = urllib.parse.quote(gcs_model_dir, safe=())
+    db.collection("models").document(model_id).set(
+        {
+            "trained_on": [_get_simulation_doc(db, sim_name) for sim_name in sim_names],
+            "trained_at_utc": datetime.datetime.utcnow(),
```
Comment (tatianawu) [inline usl_models/usl_models/flood_ml/metastore.py:33]: What's the value of this? Is this mostly to track when we saved a model?

File: usl_models/usl_models/flood_ml/metastore.py [hunk 2]
```diff
@@ -1,8 +1,42 @@
+import datetime
 from typing import Any, Sequence, TypeVar
 import urllib.parse
 
 from google.cloud import firestore  # type:ignore[attr-defined]
 
+import usl_models.flood_ml.model_params
+
+
+def write_model_metadata(
```
Comment (tatianawu) [inline usl_models/usl_models/flood_ml/metastore.py:10]: Do any of the current fields hold information about the number of epochs we trained on? If not, I think that could be useful to add too.

File: usl_models/usl_models/flood_ml/metastore.py (hunk 1 above)
Comment (waltaskew) [reply usl_models/usl_models/flood_ml/metastore.py:33]: yeah, to help distinguish the different experiments -- a better marker would be something like a commit hash or a version of the code trained on

File: usl_models/usl_models/flood_ml/metastore.py (hunk 2 above)
Comment (waltaskew) [reply usl_models/usl_models/flood_ml/metastore.py:10]: the epochs are part of the model params, which we write out

Screenshot 2024-07-19 3 11 11 PM

Comment (waltaskew) [reply usl_models/usl_models/flood_ml/metastore.py:10]: oh, wait, I'm wrong -- they used to be part of the params but no longer are! I'll add that in!

File: usl_models/usl_models/flood_ml/metastore.py (hunk 1 above)
Comment (tatianawu) [reply usl_models/usl_models/flood_ml/metastore.py:34]: What about adding something like an experiment name or description? So the trainer task could (optionally) take in a short descriptor and include that when we save the model?

File: usl_models/usl_models/flood_ml/metastore.py (hunk 2 above)
Comment (tatianawu) [reply usl_models/usl_models/flood_ml/metastore.py:10]: I think we're missing the field in the docstring.

**Review by tatianawu — APPROVED** [review]

File: usl_models/usl_models/flood_ml/metastore.py (hunk 1 above)
Comment (waltaskew) [reply usl_models/usl_models/flood_ml/metastore.py:37]: now with model name!

**Pull request merged**